localstack/services/awslambda/lambda_starter.py

```diff
@@ -19,6 +21,18 @@ def start_lambda(port=None, asynchronous=False):
     from localstack.services.awslambda import lambda_api, lambda_utils
     from localstack.services.infra import start_local_api
 
+    ROUTER.add(
+        "/",
+        host="<regex('([a-z0-9-]+\\.)?'):region><api_id>.lambda-url.<regex('.*'):server>",
+        endpoint=handle_lambda_url_invocation,
+        defaults={"path": ""},
+    )
+    ROUTER.add(
+        "/<path:path>",
+        host="<regex('([a-z0-9-]+\\.)?'):region><api_id>.lambda-url.<regex('.*'):server>",
```

Comment:
You can directly use the regex from your host pattern (instead of having a complex regex here, but actually parsing it later our of the host header).
If the region is not optional, you can directly assume the first part before the `.` is a region:
```
host="<region>.<api_id>.lambda-url.<regex('.*'):server>",
```
If it is optional, you can add two routes:
```
    ROUTER.add(
        "/<path:path>",
        host="<region>.<api_id>.lambda-url.<regex('.*'):server>",
        endpoint=handle_lambda_url_invocation,
    )
    ROUTER.add(
        "/<path:path>",
        host="<api_id>.lambda-url.<regex('.*'):server>",
        endpoint=handle_lambda_url_invocation
    )
```

This would allow you to:
- Get rid of the region parsing in the `request_context.py`
- Directly use the region and the API ID in the function signature of the endpoint function:
  ```
  def handle_lambda_url_invocation(request: Request, api_id: str, region: Optional[str] = None) ...
  ```
  Instead of:
  ```
  def handle_lambda_url_invocation(request: Request, **url_params: Dict[str, Any]) ...
  ```

Reply:
I edited the route to have a format more close to AWS so I ended up using the REGEX, I also edited the function to receive the URL parameters the way you mentioned.